tests/integration/runners/test_fileserver.py

```diff
@@ -161,6 +163,10 @@ def test_file_list(self):
         self.assertIsInstance(ret['return'], list)
         self.assertTrue('grail/scene33' in ret['return'])
 
+    # Git doesn't handle symlinks in Windows. See the thread below:
+    # http://stackoverflow.com/questions/5917249/git-symlinks-in-windows
+    @skipIf(salt.utils.is_windows(),
+            'Git doesn\'t handle symlinks properly on Windows')
```

Comment:
None of the changes you made this afternoon have any bearing on this test, since as you noted earlier, Git for Windows does not reproduce symlinks when cloning.

I opened https://github.com/saltstack/salt/pull/39746 just now, which adds a new module ``salt.utils.path`` to provide platform-independent functions to deal with symbolic links. The fileserver runner is not the place to test Windows' ability to list symlinks, because of that issue with cloning symlinks, so I also made the change to the ``skipIf`` which I recommended earlier today, to make the skip reason more clear.

The proper place to test symlink handling is in the roots fileserver backend integration tests. So, I made some modifications there, which on Windows an creates an alternate ``file_roots`` for the symlink test. I then removed the ``skipIf`` from that test and used the specially-created ``file_roots`` for that test.

Since https://github.com/saltstack/salt/pull/39746 contains the work you did on Monday, I think this pull request can be closed.